local_vars/local_vars.cc

```diff
@@ -45,18 +45,33 @@ class LocalNameInserter {
         ast::ExpressionPtr expr;
     };
 
+    // Handle the mangling of keyword argument names, if the name passed has already been seen in the argument list.
+    core::NameRef mangleKeyword(core::MutableContext ctx, const vector<NamedArg> &seen, bool isKeyword,
+                                core::NameRef name, uint32_t pos) const {
+        // Mangle any duplicate keyword arguments.
+        if (isKeyword) {
+            auto it = absl::c_find_if(seen, [name](auto &existing) { return existing.name == name; });
+            if (it != seen.end()) {
+                return ctx.state.freshNameUnique(core::UniqueNameKind::MangledKeywordArg, name, pos);
```

Comment:
The place where we mangle the names we should be reporting the error. This is how it works for e.g. mangled methods.

Mangling is a mechanism to recover from errors, and we need to ensure that that error is reported to the user, or else they'll get weird behavior and not realize what the source of it was.

Before we started making these changes, namer reported the error and namer mangled the keywords. Then, the parser did both. But now, the parser reports the error while local_vars does the mangling. That means that for any violations which are introduced between the parser and the local_vars phase (e.g. any violations in rewriter passes), we'll have missing errors.